applications/nrf_desktop/src/modules/dfu_mcumgr.c

```diff
@@ -57,42 +58,45 @@ static void dfu_timeout_handler(struct k_work *work)
 	}
 }
 
-static enum mgmt_cb_return mcumgr_img_mgmt_cb(uint32_t event,
-					      enum mgmt_cb_return prev_status,
-					      int32_t *rc, uint16_t *group,
-					      bool *abort_more, void *data,
-					      size_t data_size)
+static enum mgmt_cb_return smp_cmd_recv(uint32_t event, enum mgmt_cb_return prev_status,
+					int32_t *rc, uint16_t *group, bool *abort_more,
+					void *data, size_t data_size)
 {
-	LOG_DBG("MCUmgr Image Management Event with the %d ID",
-		u32_count_trailing_zeros(MGMT_EVT_GET_ID(event)));
+	LOG_INF("MCUmgr SMP Command Recv Event");
 
-	k_work_reschedule(&dfu_timeout, DFU_TIMEOUT);
-	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK) && dfu_lock_claim(&mcumgr_owner)) {
-		(void)k_work_cancel_delayable(&dfu_timeout);
-		*rc = MGMT_ERR_EACCESSDENIED;
-		return MGMT_CB_ERROR_RC;
-	}
+	const struct mgmt_evt_op_cmd_arg *cmd_recv;
 
 	if (IS_ENABLED(CONFIG_MCUMGR_TRANSPORT_BT) &&
 	    atomic_cas(&mcumgr_event_active, false, true)) {
 		APP_EVENT_SUBMIT(new_ble_smp_transfer_event());
 	}
```

Comment:
To follow up on this https://github.com/nrfconnect/sdk-nrf/pull/13082/files#r1392860366 comment, you can move this code block at the end of the function after the code block with the potential to return ```MGMT_ERR_EACCESSDENIED```

Reply:
Also you need to remove the following code block:

```
	if (!IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK)) {
		return MGMT_CB_OK;
	}
```